axum/src/test_helpers/counting_cloneable_state.rs

```diff
@@ -0,0 +1,59 @@
+use std::sync::{
+    atomic::{AtomicBool, AtomicUsize, Ordering},
+    Arc, Mutex,
+};
+
+pub(crate) struct CountingCloneableState {
+    state: Arc<Mutex<InnerState>>,
+}
```

Comment:
Nice 👍🏼

Reply:
actually, we don't need a `Mutex` as we're using atomic values: https://github.com/tokio-rs/axum/pull/2970/commits/d9ef8e02c7087d5bd52605ffc23dc008ccfaaea5